apiserver/facades/client/client/status.go

```diff
@@ -1098,10 +1102,18 @@ func (c *statusContext) makeMachineStatus(
 	sModInfo, err := c.status.MachineModification(machineID)
 	populateStatusFromStatusInfoAndErr(&status.ModificationStatus, sModInfo, err)
 
-	instid, displayName := c.allInstances.InstanceNames(machineID)
+	var instid string
+	machineUUID, err := machineService.GetMachineUUID(ctx, coremachine.Name(machineID))
+	if err != nil {
+		logger.Debugf("error retrieving uuid for machine: %q, %w", machineID, err)
+	} else {
+		instid, err = machineService.InstanceID(ctx, machineUUID)
+		if err != nil && !errors.Is(err, machineerrors.NotProvisioned) {
+			logger.Debugf("error retrieving instance ID for machine: %q, %w", machineID, err)
+		}
+	}
 	if instid != "" {
-		status.InstanceId = instid
-		status.DisplayName = displayName
+		status.InstanceId = instance.Id(instid)
```

Comment:
```suggestion
		status.InstanceId = instance.Id(instid)
```

If your service a instance.Id, this wouldn't be required. It has cropped up a few times.